File: marsrover/src/main/direction/Direction.cpp
#include <bits/stdc++.h>
#include "../rover/Rover.cpp"
#include "DirectionVisitor.cpp"
using namespace std;

class Direction
{
public:
    virtual Direction spinRight() = 0;
    virtual Direction spinLeft() = 0;
    virtual void moveForward(Rover rover) = 0;
    virtual void moveBack(Rover rover) = 0;
    virtual char getDirectionChar() = 0;
};

File: marsrover/src/main/java/com/thoughtworks/direction/Direction.java
package java.com.thoughtworks.direction;

import java.com.thoughtworks.rover.Rover;

public interface Direction {

    public abstract Direction spinRight();

    public abstract Direction spinLeft();

    public abstract void moveForward(Rover rover);

    public abstract void moveBackward(Rover rover);

    public abstract char getDirectionChar();
};
